src/lib/Form/Type/Content/ContentFieldType.php

```diff
@@ -48,6 +48,8 @@ public function configureOptions(OptionsResolver $resolver)
                 'location' => null,
                 'contentCreateStruct' => null,
                 'contentUpdateStruct' => null,
+                'userCreateStruct' => null,
+                'userUpdateStruct' => null,
```

Comment:
I am not exactly happy that we're introducing yet another optional option, from which one of those should actually be set. The fact that this struct (which is the form parent data type) was missing was actually how this issue got through our checks in the first place.

Therefore, I'd suggest:
 * Adding an expectation that those structs need to be of specific data type ([`OptionsResolver::setAllowedTypes()`](https://symfony.com/doc/5.x/components/options_resolver.html#type-validation).
 * Putting all those options under a single option, with proper allowed types, making it required, and deprecating all the other ones. To maintain backward compatibility, we would add a [Option Normalizer](https://symfony.com/doc/5.x/components/options_resolver.html#option-normalization) to move the options under old names to a new one.
 * Or, alternatively, research how we could enforce that exactly one of those options is set.

That would also work as a safeguard to prevent this issue from coming back, if we ever add more structs.

Reply:
@Steveb-p 
> we would add a [Option Normalizer](https://symfony.com/doc/5.x/components/options_resolver.html#option-normalization) to move the options under old names to a new one.

I'm not sure we can convert old name options like `contentUpdateStruct` to `struct`  option using `Option Normalizer`. Rather we could convert a new `struct`'s value to another option's value i.e. `contentUpdateStruct`'s value. Nevertheless, please see the latest iteration, not sure if this is the direction you would like to go for.

Reply:
Your changes are roughly what I had in mind. You are in fact converting all the "old" options into a new, singular one.

If it works as expected (puts old values into the new option), then you're mostly done and just need to adjust the code to be more "strict".

For example, we should not overwrite a non-null value in the option.

We should ensure that `content_update_struct` can only contain `ContentUpdateStruct` class objects.

Things like that.